talentmap_api/common/common_helpers.py

```
@@ -110,6 +112,39 @@
     "Vietnamese": "Vietnamese",
 }
 
+def view_result(result):
+    '''
+    Returns view result with correct error code
+    '''
+    code = status.HTTP_200_OK
+    if (result.get('return_code') is -1):
+        code = status.HTTP_400_BAD_REQUEST
+    if (result.get('return_code') is -2):
+        code = status.HTTP_500_INTERNAL_SERVER_ERROR
```

the `-1` makes sense ... but, are we sure that `-2` should return a 500 error?

i TOTALLY might be wrong here - but i would think we want a return code of `0` to return a 200 OK, `-1` to be a 400 Bad Request and everything else to be a generic 500 🤔

Just a thought based on what i can remember off the top of my head ... this will likely be a work in progress anyways

---

Based on the emails I got, the only return possibilities are 2, 1, 0, -1, -2.

I 100% get you and I honestly need a bit of help deciding what error code to return for these too. We might be better off just returning a 400 or 500 overall and just include error messages in the response to assist with dev/testing- and then later on when we have a more definitive pattern for what errors return what, we can redefine this block. Or I can just leave this as well.

This is what WS sent me:

-1  - Failed transaction with User-defined error message
-2  - Failed transaction with undefined usually catastrophic error message
0 - Successful transaction with no messages
+1 - Successful transaction with User-defined warning messages
+2 - Successful transaction with User-defined information (typical for Batch processes to capture metrics)

---

oh wow - if that's what they sent you, then I'd go with that. 

the +1 & +2 would just be other forms of a 200 response (so no need to handle them imo - also, i don't think I've seen one returned yet, this could be something we add on later once we start seeing the response) 

the -1 & -2 you have right as is.

so i think this is OK

---

Based on what Jeanie said - it doesn't sound like a -1 or -2 changes what the error could be (just how they return info).

I'd say lets use: 
0 - 200
Not 0 - 400
No error code - 500 

But for -1 we can log their msg in our error log, and for -2 just log generic error msg in our error log. Generally we don't want to couple our services too tightly with their stuff. Their errors we assume are for devs not for users. Their errors should inform our API logic to ideally prevent them from happening